SU2_CFD/src/output/CMultizoneOutput.cpp

```diff
@@ -125,6 +125,41 @@ void CMultizoneOutput::SetMultizoneHistoryOutputFields(const COutput* const* out
         AddHistoryOutput(name, header, field.screenFormat, group, "", field.fieldType );
       }
     }
+    /*--- Also add the PerSurface outputs ---*/
+    const auto& ZoneHistoryPerSurfaceFields = output[iZone]->GetHistoryPerSurfaceFields();
+
+    vector<string> Marker_Analyze;
+    for (unsigned short iMarker_Analyze = 0; iMarker_Analyze < config[iZone]->GetnMarker_Analyze(); iMarker_Analyze++){
+      Marker_Analyze.push_back(config[iZone]->GetMarker_Analyze_TagBound(iMarker_Analyze));
+    }
+    //Marker_Analyze.push_back("test");
+
+    for (const auto& nameSinglezone : output[iZone]->GetHistoryOutputPerSurface_List()) {
+
+      if (nameSinglezone != "TIME_ITER" && nameSinglezone != "OUTER_ITER") {
+
+        const auto& field = ZoneHistoryPerSurfaceFields.at(nameSinglezone);
+
+        name   = nameSinglezone + zoneIndex;
+
+        // field[0].fieldName contains names like Avg_Massflow(fluid_outlet) but looking at CFlowOutput.cpp it is also:
+        // AddHistoryOutputPerSurface("SURFACE_MASSFLOW",         "Avg_Massflow", ... without the markerTag already attached
+        // So first the fieldName for one of the markers (here the 0th) is taken and the markerTag in brackets is removed
+        string baseheader = field[0].fieldName;
+        std::string::size_type pos = baseheader.find('(');
+        if (pos != std::string::npos)
+            baseheader = baseheader.substr(0, pos);
+        else
+            SU2_MPI::Error("Cannot proccess PerSurface *_SURF history output.", CURRENT_FUNCTION);
+
+        header = baseheader + zoneIndex; // field[i] where i is the marker
+        group  = field[0].outputGroup + zoneIndex;
+
+        AddHistoryOutputPerSurface(name, header, field[0].screenFormat, group, Marker_Analyze, field[0].fieldType );
+        //AddHistoryOutputPerSurface("SURFACE_MASSFLOW", "Avg_Massflow", ScreenOutputFormat::SCIENTIFIC, "FLOW_COEFF_SURF", Marker_Analyze, HistoryFieldType::COEFFICIENT);
```

Comment:
I tried to go with how it is done for single zone plus changes that are also done for the non-per-Surface things.
As of now, the _SURF fields are recognized but all values in the history file are zero. I am not exactly sure how everything works in the output-machine-room but I currently fear that there is a mismatch somewhere in naming maybe. I guess the convention like e.g. `Avg_Massflow[0](fluid_outlet)` is best

Reply:
This bit is still a bit hacked, get it working first and nice second

Reply:
working in 7ba1efc